### PR TITLE
[DOC-10592] Add details about NUMA

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-10592] Add details about NUMA

### CODE REVIEW

File: src/current/_includes/common/numa_and_go.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+In a [NUMA (non-uniform memory access) architecture](https://en.wikipedia.org/wiki/Non-uniform_memory_access), the system’s memory is physically distributed across multiple memory banks or "nodes", and each node is assigned to a processor. A processor can access its local memory much faster than non-local memory. This non-uniform memory access can lead to performance differences depending on data location.
+
+CockroachDB is written in Go, which has no support for NUMA scheduling or pinning. To run multiple CockroachDB clusters on a NUMA architecture:
+
+- Assign no more than 32 cores to the node for each instance.
+- Ensure that your orchestration framework, process manager, or startup scripts start each CockroachDB in a separate NUMA node.
+- Use the `--locality` flag to indicate when multiple CockroachDB instances are running on a given physical host.
+- Ensure that each CockroachDB cluster writes to a unique set of storage volumes (block devices).
```
Comment (sean-) [inline src/current/_includes/common/numa_and_go.md:8]: I don't understand this last recommendation relative to the rest of the recommendations and would remove it from this PR.  i.e., why is the storage volume important here?  I don't think we've made any recommendations about aligning storage devices and NICs to the same NUMA domain, so I don't want to expand the scope of these guidelines into general best practices for NUMA.

File: src/current/_includes/common/numa_and_go.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+In a [NUMA (non-uniform memory access) architecture](https://en.wikipedia.org/wiki/Non-uniform_memory_access), the system’s memory is physically distributed across multiple memory banks or "nodes", and each node is assigned to a processor. A processor can access its local memory much faster than non-local memory. This non-uniform memory access can lead to performance differences depending on data location.
+
+CockroachDB is written in Go, which has no support for NUMA scheduling or pinning. To run multiple CockroachDB clusters on a NUMA architecture:
+
+- Assign no more than 32 cores to the node for each instance.
```
Comment (sean-) [inline src/current/_includes/common/numa_and_go.md:5]: > See also, [`numactl(8)`](https://man7.org/linux/man-pages/man8/numactl.8.html)'s `--membind` flag. 

?

**Review by sean- — APPROVED** [review]

File: src/current/_includes/common/numa_and_go.md (hunk 1 above)
Comment (mdlinville) [reply src/current/_includes/common/numa_and_go.md:8]: I took all of this straight from what @nicktrav had there. Maybe he can weigh in.

File: src/current/_includes/common/numa_and_go.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+- Ensure that your orchestration framework, process manager, or startup scripts start each CockroachDB in a separate NUMA node.
+- Use the `--locality` flag to indicate when multiple CockroachDB instances are running on a given physical host.
```
[the remark this answers is not in the document]
Comment (mdlinville) [reply src/current/_includes/common/numa_and_go.md]: Done

File: src/current/_includes/common/numa_and_go.md (hunk 2 above)
Comment (mdlinville) [reply src/current/_includes/common/numa_and_go.md]: Done

File: src/current/_includes/common/numa_and_go.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+In a [NUMA (non-uniform memory access) architecture](https://en.wikipedia.org/wiki/Non-uniform_memory_access), the system’s memory is physically distributed across multiple memory banks or "nodes", and each node is assigned to a processor. A processor can access its local memory much faster than non-local memory. This non-uniform memory access can lead to performance differences depending on data location.
+
+CockroachDB is written in Go, which has no support for NUMA scheduling or pinning. To run multiple CockroachDB clusters on a NUMA architecture:
```
Comment (sean-) [inline src/current/_includes/common/numa_and_go.md:3]: > which has no support for NUMA scheduling or planning

change to:

> which has no native support for NUMA scheduling or planning

Small nit: Go can't constrain itself to a NUMA node, but the OS can.

**Review by sean- — APPROVED** [review]

File: src/current/_includes/common/numa_and_go.md (hunk 4 above)
Comment (mdlinville) [inline src/current/_includes/common/numa_and_go.md:3]:
```suggestion
CockroachDB is written in Go, which has no process-level support for NUMA scheduling or pinning. Instead, you must manage NUMA at the operating system level. To run multiple CockroachDB clusters on a NUMA architecture:
```

**Comment by mdlinville** [comment]
I'll go ahead and merge this. Thanks for the review!

**Pull request merged**